generator/src/main/java/com/scottlogic/deg/generator/walker/ReductiveWalkerRetryChecker.java

```diff
@@ -0,0 +1,27 @@
+package com.scottlogic.deg.generator.walker;
+
+public class ReductiveWalkerRetryChecker {
+    private int numRetriesSoFar = 0;
+    private boolean guaranteedNotFullyContradictory = false;
```

Comment:
what's the purpose of this ```guaranteedNotFullyContradictory``` variable?

Reply:
As soon as you can successfully generate a value, you know that the profile is not fully contradictory, and so the retry checker is not needed. This variable reflects that. Do you have a suggestion to make it clearer?

Reply:
2 things.

1. in random mode, it restarts every time anyway, so this is not needed.

2. if it didn't restart, it's not guaranteed that generating one value means that the tree cannot end up in a contradictory infinite loop state.